File: server/Application/DTOs/TripResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using Domain.Enums;

namespace Application.DTOs
{
    public class TripResponse
    {
        public string Title { get; set; }

        public string? Description { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public TripStatus? Status { get; set; }
        public List<ImageResponse> Images { get; set; } = new List<ImageResponse>();
        public LocationDTO Location { get; set; }
        public UserResponse User { get; set; }
        public RatingResponse Rating { get; set; }
        public List<TripActivityResponse> TripActivities { get; set; }
       
    }
}

File: client/src/types/TripResponse.ts
import { TripStatus } from "./enums/TripStatus";
import { ImageResponse } from "./ImageResponse";
import { LocationDTO } from "./LocationDTO";
import { RatingResponse } from "./RatingResponse";
import { TripActivityResponse } from "./TripActivityResponse";

export interface TripResponse {
  id: number;
  title: string;
  description: string;
  startDate: Date;
  endDate: Date;
  status: TripStatus;
  images: ImageResponse[];
  location: LocationDTO;
  rating: RatingResponse;
  tripActivities: TripActivityResponse[];
  user: {
    FirstName: string;
    LastName: string;
  };
  isArchive: boolean;
}
